resolwe_bio/processes/slamdunk/slam_count.py

```diff
@@ -0,0 +1,121 @@
+from plumbum import TEE
+
+from resolwe.process import Cmd, DataField, FileField, JsonField, Process, StringField
+
+import pandas as pd
+
+
+def prepare_expressions(infile, rc_file, tpm_file):
+    """Compute TPM values from T>C read counts."""
+    exp = pd.read_csv(
+        infile,
+        sep='\t',
+        usecols=['gene_name', 'length', 'tcReadCount'],
+        index_col='gene_name',
+        dtype={
+            'gene_name': str,
+            'length': int,
+            'tcReadCount': int,
+        },
+    )
+    exp['rpk'] = exp.apply(lambda x: (exp['tcReadCount'] * 1e3 / exp['length']), axis=1)
+    rpk_sum = exp['rpk'].sum()
+    exp['tpm'] = exp.apply(lambda x: (exp['rpk'] / rpk_sum * 1e6), axis=1)
+    rc = exp['tcReadCount'].to_csv(rc_file, index_label='Gene', header=['Expression'], sep='\t', compression='gzip')
+    tpm = exp['tpm'].to_csv(tpm_file, index_label='Gene', header=['Expression'], sep='\t', compression='gzip')
+    return (rc, tpm)
+
+
+class SlamCount(Process):
+    """Convert gene-level Slamdunk data to the expressions data type."""
+
+    slug = 'slam-count'
+    process_type = 'data:expression:slamdunk'
+    name = 'Slam count'
+    requirements = {
+        'expression-engine': 'jinja',
+        'executor': {
+            'docker': {
+                'image': 'resolwebio/slamdunk:1.0.0'
+            },
+        },
+        'resources': {
+            'cores': 1,
+            'memory': 1024,
+            'network': True,
+        },
+    }
+    category = 'Slamdunk'
+    entity = {
+        'type': 'sample',
+    }
+    data_name = '{{ tcount|sample_name|default("?") }}'
+    version = '1.0.0'
+
+    class Input:
+        """Input fields for SlamCount."""
+
+        tcount = DataField('alleyoop:collapse', label='Collapsed Slamdunk results')
+        source = StringField(
+            label="Gene ID source",
+            default='ENSEMBL',
+            choices=[
+                ('ENSEMBL', 'ENSEMBL'),
+                ('UCSC', 'UCSC'),
+            ],
+        )
+
+    class Output:
+        """Output fields to process SlamCount."""
+
+        exp = FileField(label="Normalized expression")
+        exp_json = JsonField(label="Expression (json)")
+        exp_type = StringField(label="Expression type")
+        rc = FileField(label="T>C read counts")
+        exp_set = FileField(label="Expressions")
+        exp_set_json = JsonField(label="Expressions (json)")
+        source = StringField(label="Gene ID source")
+        species = StringField(label="Species")
+        build = StringField(label="Build")
+        feature_type = StringField(label="Feature type")
+
+    def run(self, inputs, outputs):
+        """Run analysis."""
+        basename = os.path.basename(inputs.tcount.tcount.path)
+        assert basename.endswith('.txt')
+        name = basename[:-4]
+
+        rc_file = name + '_rc.txt.gz'
+        tpm_file = name + '_tmp.txt.gz'
+
+        prepare_expressions(inputs.tcount.tcount.path, rc_file, tpm_file)
```

Comment:
Perhaps a check that would make sure the files have been created? This could already be done within the `prepare_expressions` function.